framework/src/play-filters-helpers/src/main/scala/play/filters/https/RedirectHttpsFilter.scala

```diff
@@ -49,10 +49,13 @@ class RedirectHttpsFilter @Inject() (config: RedirectHttpsConfiguration) extends
       next(req).map(_.withHeaders(stsHeaders: _*))
     } else {
       val xForwarded = forwardedProto(req)
-      if (redirectEnabled && xForwarded) {
+      val isExcludePath = checkExcludePath(req)
+      if (redirectEnabled && xForwarded && !isExcludePath) {
         Accumulator.done(Results.Redirect(createHttpsRedirectUrl(req), redirectStatusCode))
+      } else if (isExcludePath) {
+        Accumulator.done(Results.Status(200))
       } else {
-        if (xForwarded) {
+        if (xForwarded && !isExcludePath) {
           logger.info(s"Not redirecting to HTTPS because $redirectEnabledPath flag is not set.")
```

Comment:
Should log at debug level:

```suggestion
          logger.debug(s"Not redirecting to HTTPS because $redirectEnabledPath flag is not set.")
```

Reply:
@wsargent 
I fixed it.
https://github.com/playframework/playframework/pull/8817/files#diff-8418f5188b6f93386cfd89ce9f9e82c3R59